serviceAccount/ServiceAccount.psm1

```diff
@@ -0,0 +1,1546 @@
+        }
+        else
+        {
+             Write-Host "Service account $ServiceAccount with SID $SID was not found in the Authorization Policy rule set"
+        }
+        $Properties.PolicyStore.AuthorizationPolicy = $AuthorizationPolicyRules.ToString()
+
+        $AuthorizationPolicyReadOnlyRules = [AdfsRules]::new($Properties.PolicyStore.AuthorizationPolicyReadOnly) 
+        if($AuthorizationPolicyReadOnlyRules.RemoveRule($ServiceAccountRule))
+        {
+            Write-Host "Service account $ServiceAccount with SID $SID was removed from the Authorization Policy Read Only rule set"
+        }
+        else
+        {
+            Write-Host "Service account $ServiceAccount with SID $SID was not found in the Authorization Policy Read Only rule set"
+        }
+        $Properties.PolicyStore.AuthorizationPolicyReadOnly = $AuthorizationPolicyReadOnlyRules.ToString()
+
+    }
+    else
+    {
+        #Check if rule already exists in auth policy
+        $AuthorizationPolicyRules = [AdfsRules]::new($Properties.PolicyStore.AuthorizationPolicy)
+        if($AuthorizationPolicyRules.FindIndexForRule($ServiceAccountRule) -ne -1)
+        {
+            Write-Host "Service account rule already exists."
+            return
+        }
+        Write-Host "Adding rule for service account $ServiceAccount with SID $SID to Authorization Polixy and Authorization Policy Read Only rule sets"
+
+        $Properties.PolicyStore.AuthorizationPolicy = $Properties.PolicyStore.AuthorizationPolicy + $ServiceAccountRule
+        $Properties.PolicyStore.AuthorizationPolicyReadOnly = $Properties.PolicyStore.AuthorizationPolicyReadOnly + $ServiceAccountRule
+    }
+
+    Set-AdfsInternalSettings  (Get-DataContractSerializedString -object $Properties) | out-null
+
+
+    $doc = new-object Xml
+    $doc.Load("$env:windir\ADFS\Microsoft.IdentityServer.Servicehost.exe.config")
+    $connString = $doc.configuration.'microsoft.identityServer.service'.policystore.connectionString
+
+
+    if((IsWID -ConnectionString $connString) -eq $true)
+    {
+        if($SecondaryServers.Count -eq 0)
+        {
+            Write-Warning("No list of secondary servers was provided. You must ensure a sync has occurred on all machines before proceeding to change the service account.")
+        }
+
+        #In the case of WID, sync config among all secondary servers
+        foreach($Server in $SecondaryServers)
+        {
+            Invoke-Command -ComputerName $Server -ScriptBlock {
+                $Date = Get-Date 
+                $Duration = (Get-AdfsSyncProperties).PollDuration
+                Set-AdfsSyncProperties -PollDuration 1
+                while((Get-AdfsSyncProperties).LastSyncTime -lt $Date)
+                {
+                    Start-Sleep 1
+                }
+                Set-AdfsSyncProperties -PollDuration $Duration
+            }
+        }
+    }
+
+} 
+
+Function UserWantsToContinue
```

Comment:
Can this function be removed?

Reply:
Why would this function be removed? It's used in both Update-AdfsServiceAccountRule and Restore-ADFSSettingsFromBackup

Reply:
Nvm...looks like this change was missed in a previous commit